app/src/main/java/com/android/feedme/MainActivity.kt

```diff
@@ -59,11 +62,16 @@ class MainActivity : ComponentActivity() {
                 }
 
                 navigation(startDestination = Screen.HOME, route = Route.HOME) {
-                  composable(Screen.HOME) { LandingPage(navigationActions) }
-                  composable(Screen.RECIPE) { backStackEntry ->
-                    backStackEntry.arguments?.getString("recipeId")?.let {
-                      RecipeFullDisplay(navigationActions)
-                    }
+                  composable(Screen.HOME) {
+                    // Create a shared view model for Recipe
+                    val recipeViewModel = viewModel<RecipeViewModel>()
+                    LandingPage(navigationActions, recipeViewModel)
+                  }
+                  composable(Screen.RECIPE) {
+                    // Link the shared view model to the composable
+                    val navBackStackEntry = navController.getBackStackEntry(Screen.HOME)
+                    val recipeViewModel = viewModel<RecipeViewModel>(navBackStackEntry)
```

Comment:
Duplicate of line 67, they are the same instance and will die if we leave `Screen.HOME`.
However, another is an issue, as the `RecipeFullDisplay` may be access from the Profile, do we re-create another `RecipeViewModel` for only that Route or should we share it with` Screen.HOME` (i.e a single `RecipeViewModel` for the whole app) ?

Reply:
We discussed privately and came to the conclusion that this case will be dealt later when implemented.